app/server/appsmith-server/src/main/java/com/appsmith/server/filters/PreAuth.java

```
@@ -20,41 +20,41 @@ public class PreAuth implements WebFilter {
+    private final String USERNAME = "username";
 
     public PreAuth(RateLimitService rateLimitService) {
         this.rateLimitService = rateLimitService;
     }
 
     @Override
     public Mono<Void> filter(ServerWebExchange exchange, WebFilterChain chain) {
-        log.debug("Inside PreAuth filter");
         return getUsername(exchange).flatMap(username -> {
-            if (username != null) {
-                log.debug("Username: {}", username);
+            if (!username.isEmpty()) {
                 return rateLimitService
                         .tryIncreaseCounter(RateLimitConstants.BUCKET_KEY_FOR_LOGIN_API, username)
                         .flatMap(counterIncreaseAttemptSuccessful -> {
                             if (!counterIncreaseAttemptSuccessful) {
                                 log.error("Rate limit exceeded. Redirecting to login page.");
                                 return handleRateLimitExceeded(exchange);
-                            } else {
-                                return chain.filter(exchange);
                             }
+
+                            return chain.filter(exchange);
                         });
             } else {
-                log.error("Username is empty. Continuing with filter chain.");
+                // If username is empty, simply continue with the filter chain
                 return chain.filter(exchange);
             }
         });
     }
 
     private Mono<String> getUsername(ServerWebExchange exchange) {
         return exchange.getFormData().flatMap(formData -> {
-            String username = formData.getFirst("username");
-            if (username != null) {
+            String username = formData.getFirst(USERNAME);
+            if (username != null && !username.isEmpty()) {
                 return Mono.just(URLDecoder.decode(username, StandardCharsets.UTF_8));
             }
-            return Mono.empty();
+
+            return Mono.just("");
```

Should we throw an error here? username being an empty string should not be allowed considering username is a unique identifier in our system.

---

@trishaanand it will not be allowed in the authentication flow. It's just for the rate limit filter. I've tested the flow